cpu/atmega2560/periph/gpio.c

```diff
@@ -28,48 +27,500 @@
 
 #if GPIO_NUMOF
 
+/*
 typedef struct {
     void (*cb)(void);
 } gpio_state_t;
+*/
 
-/*static gpio_state_t config[GPIO_NUMOF]; */
-
+/* static gpio_state_t config[GPIO_NUMOF]; */
 
+/* static pin mapping */
+static const uint8_t gpio_pin_map[GPIO_NUMOF] = {
+#if GPIO_0_EN
+    [GPIO_0] = GPIO_0_PIN,
+#endif
+#if GPIO_1_EN
+    [GPIO_1] = GPIO_1_PIN,
+#endif
+#if GPIO_2_EN
+    [GPIO_2] = GPIO_2_PIN,
+#endif
+#if GPIO_3_EN
+    [GPIO_3] = GPIO_3_PIN,
+#endif
+#if GPIO_4_EN
+    [GPIO_4] = GPIO_4_PIN,
+#endif
+#if GPIO_5_EN
+    [GPIO_5] = GPIO_5_PIN,
+#endif
+#if GPIO_6_EN
+    [GPIO_6] = GPIO_6_PIN,
+#endif
+#if GPIO_7_EN
+    [GPIO_7] = GPIO_7_PIN,
+#endif
+#if GPIO_8_EN
+    [GPIO_8] = GPIO_8_PIN,
+#endif
+#if GPIO_9_EN
+    [GPIO_9] = GPIO_9_PIN,
+#endif
+#if GPIO_10_EN
+    [GPIO_10] = GPIO_10_PIN,
+#endif
+#if GPIO_11_EN
+    [GPIO_11] = GPIO_11_PIN,
+#endif
+#if GPIO_12_EN
+    [GPIO_12] = GPIO_12_PIN,
+#endif
+#if GPIO_13_EN
+    [GPIO_13] = GPIO_13_PIN,
+#endif
+};
 
+/*
+ * TODO: Implement pushpull
+ */
 int gpio_init_out(gpio_t dev, gpio_pp_t pushpull)
 {
-    return -1;
+    uint8_t pin;
+
+    if (dev > GPIO_NUMOF || dev < 2)
+    {
+        return -1; /* Can't initialize pin */
+    }
+
+    pin = gpio_pin_map[dev];
+
+    switch (dev)
+    {
+        case 0:
```

Comment:
Would be nice and better readable if you substitute `case 0` by `case GPIO_0` and following